### ISSUE TITLE
Remove/combine django views

### ISSUE THREAD

**Issue by cmc333333** [18F/atf-eregs#207]
We have a rather large number of views (e.g. `chrome_section_view`, `ChromeSubterpView`, `partial_regulation_view`, etc. etc.)

This is really hard to follow and makes url generation very complicated. Pare it down -- we probably just need a `chrome` function to wrap partial end points and we can remove the distinction between most section/subterp/regulation/paragraph/etc.

**Comment by jbarnicle** [comment]
I'd like to entertain the thought of dropping chrome* as a naming convention.  I don't know that the intended etymology is widely understood (wasn't to me) and there is a large potential for confusion with the browser (again - was to me).

**Comment by cmc333333** [comment]
:+1: We'll need a better term, but I think this'll boil down to something like views for

* Regulation
* Diff
* Search
* Breakaway

which render the main content of the page (referred to as "partials" elsewhere), and a wrapper which generates the chrome markup. Something like
```wrap_with_chrome(RegulationView)```

**Comment by jmcarp** [comment]
@cmc333333: I may be misunderstanding (wouldn't be the first time), but since many of the chrome view classes override methods of `ChromeView`, I'm not sure how a generic chrome wrapper would work--I'm assuming we want to keep all the current behavior, but make the code easier to read. I could imagine consolidating `PartialView` and `ChromeView` into a single base class, like `WrappableView`, and mounting views like this:

```
url(r'^diff/%s/%s/%s % (...)$', DiffWrappableView.as_view(), {'wrap': True}),
url(r'^partial/diff/%s/%s/%s % (...)$', DiffWrappableView.as_view(), {'wrap': False}),
```

Wrappable views could define the intersection of all methods defined by `ChromeView` and `PartialView`, or possibly compose two separate classes, `partial_class` and `wrapper_class`, neither of which would know about requests or responses. Wrappable views would then check the `wrap` flag, then delegate context creation to the composed classes as appropriate.

Let me know if anything along these lines makes sense, or if you had something different in mind.

**Comment by cmc333333** [comment]
@jmcarp I think there are a few challenges here, and I apologize for not being more detailed.

* `partial_definition_view`, `partial_section_view`, `partial_subterp_view`, `partial_interp_view`, `partial_regulation_view`, `partial_paragraph_view` feel very redundant and make constructing a url (e.g. via `reverse()`) annoying as all get-out.
* although chrome views have lots of little overridden methods, I feel like there's probably a way to combine all of them better -- I know there's a different template for diff-related-chrome and everything-else-chrome, but other distinctions seem less important (arguably even that is rather repetitive). Ideally we could remove those little nuances or at the very least, have them derived _within_ the view class
* the construction of "partial" views (including the sidebar) within chrome views is rather nasty -- we make a fake request, render the results, then shove the response's contents into a template. There's got to be a better solution
* we originally expected to need to AJAX in arbitrary sidebars, but that hasn't actually arisen. It'd probably make more sense to just include them with the "partial" view. That said, maybe combining is the wrong approach (given the next bit:)

Re: the `wrap` idea -- this is interesting, but I think we want the other direction, _decoupling_ the partial from the chrome versions. I suspect we'll want to wrap arbitrary paragraphs/diffs/etc. with new content for N&C -- from the other perspective, we'll want to embed those paragraphs/etc. in different chrome.

Above all else, though, the intent here is to get you familiar with these systems, warts and all, so you'll be prepared for integrating some or all of it into N&C.

**Comment by jmcarp** [comment]
I don't disagree with any of the above, and I'll continue figuring out how the current code works--agreed that it's more useful to understand the design here than to start changing stuff. Just to finish a thought from earlier, I think it could be interesting to compose pairs of nested views:

```python
class InterpView(ComposedView):
    parent_view = InterpWrapper
    child_view = InterpPartial
```

Or a class factory that takes a sequence of views:

```python
InterpView = compose(InterpWrapper, InterpPartial)
```

The idea would be to make the composed views Django-agnostic--they might take a dict of request parameters and return a tuple of (template, context). Requests and responses would be handled by the base class or class factory, so we wouldn't have to deal with fake requests (which, by the way, might be inelegant, but IMO isn't a major barrier to clarity).

Not pushing any of this, just saying thoughts on the chance that something winds up being useful. If we can simplify the chrome views, this line of thinking might unhelpful.
